tools/gmhazard_scripts/db_creation/empirical_db/new_calc_emp_ds.py

```diff
@@ -0,0 +1,260 @@
+                )
+                GMMs = empirical_factory.determine_all_gmm(
+                    classdef.Fault(tect_type=classdef.TectType[tect_type]),
+                    str(im),
+                    model_dict,
+                )
+                for GMM_idx, (GMM, _) in enumerate(GMMs):
+                    # TODO: Check those models
+                    if GMM.name in ("K_20", "K_20_NZ", "ZA_06", "CB_14", "ASK_14"):
+                        print(f"{GMM.name} is currently not supported.")
+                        continue
+
+                    db_type = f"{GMM.name}_{tect_type}"
+                    # Create a DB if not exists
+                    imdb = gc.dbs.IMDBParametric(
+                        str(
+                            Path(output_dir)
+                            / gc.utils.create_parametric_db_name(
+                                db_type, gc.constants.SourceType.distributed, suffix
+                            )
+                        ),
+                        writeable=True,
+                        source_type=gc.constants.SourceType.distributed,
+                    )
+                    print(
+                        f"Processing Model: {GMM.name} for {tect_type}, {GMM_idx + 1} / {len(GMMs)}"
+                    )
+                    with imdb as imdb:
+                        for site in sites:
+                            rupture_context_df = create_rupture_context_df(
+                                fault_df.merge(
+                                    distance_store.station_data(site),
+                                    left_on="fault_name",
+                                    right_index=True,
+                                ),
+                                site_df.loc[site],
+                                nhm_data,
+                                classdef.TectType[tect_type],
+                            )
+                            gmm_calculated_df = openquake_wrapper_vectorized.oq_run(
+                                GMM,
+                                classdef.TectType[tect_type],
+                                rupture_context_df,
+                                str(im),
+                                psa_periods if im is gc.im.IMType.pSA else None,
+                            )
+                            # Matching the index with rupture_df
+                            # to have a right rupture label
+                            gmm_calculated_df.set_index(
+                                rupture_df[
+                                    rupture_df["rupture_name"].isin(
+                                        rupture_context_df["rupture_name"]
+                                    )
+                                ].index,
+                                inplace=True,
+                            )
+
+                            # Relabel the columns
+                            # PGA_mean -> PGA
+                            gmm_calculated_df.columns = [
```

Comment:
@claudio525 Do you think it's better to have one for loop than two list comprehensions?
I'm not sure which is cleaner.

Reply:
I personally think I'd prefer one for loop?

Reply:
I was in that dilemma. However, even if we support all IMs we have, columns will be smaller than 200 and apparently, list comprehension seems faster in this kind of simple case as well. https://stackoverflow.com/questions/44666215/single-list-iteration-vs-multiple-list-comprehensions

Hence, I chose list comprehensions over one for loop.

Reply:
Magic. :exploding_head: 

Since that's the case, I'd suggest making a function to remove the duplicate code.

Should just need the df, a search string, and an optional suffix

Reply:
Does it always have the the `_mean` and `std suffix`? If thats the case you can also do it with np.char functionality (i.e. vectorise it)
Also, instead of splitting the whole string, you can just use `rsplit` and set maxsplit then you don't need a join.

In terms of for loop vs list comprehension, I think in this case either or seems fine, I would probably use list comprehensions.

Reply:
Actually, instead of using split, you could also use rstrip, e.g. for the `mean` case:
```
gmm_calculated_df.columns = np.char.rstrip(gmm_calculated_df.columns.values.astype(str), "_mean")
```
is all you need I think? (I didn't test it, so no guarantees haha)

Reply:
@claudio525 CC(@jasonmotha) Yeah, it does always have `_mean` and `_std_suffix` and managed to update those with magic numpy methods, so we probably don't need to create a small helper function as one is using `rstrip` and another one is using `replace`